### PR TITLE
Fix interactive documentation

### PR DESCRIPTION
**Pull request by samedii**
This is a draft but it is working and mostly done. I would like your input on the changes while I work on updating the sphinx docs and creating an example with docker-compose + fastapi + keycloak.

Some changes from the top of my head:
- I made issuer and audience both optional, is that a stupid option? Is it very insecure to not verify the issuer?
- Client id is never used so I removed it
- Do you know if `client_secret` should be provided in our service for some flows? I've seen it used in some other libs

Some grant flows are not possible to do in swagger but I'm very happy that so many are working. I've only seen "implicit" flow supported before. (Implicit flow is slightly lower security...)

### CODE REVIEW

File: fastapi_oidc/types.py [hunk 1]
```diff
@@ -1,17 +1,9 @@
 from typing import List
 
 from pydantic import BaseModel
-from pydantic import BaseSettings
 from pydantic import Extra
 
 
-class OIDCConfig(BaseSettings):
```
Comment (HarryMWinters) [inline fastapi_oidc/types.py:8]: Yikes! That should have been BaseModel 😬

**Review by HarryMWinters — CHANGES_REQUESTED** [review]
There's lots of good stuff in this PR @samedii. In particular, the improvements to the documentation, the new workflow, and the bug fix itself are all stellar. Unfortunately, we are limited by the API I defined when I published. I don't like all of the past decisions I made but we can't break backward compatibility. At least not without a major version bump and even then I'm hesitant.

How about this? We break this PR into multiple smaller PRs. 
- One to address the URL configuration and interactive FastAPI docs (this PR).
- Another PR to update the documentation (and add examples).
- Another to address issues like whether we should require `client_id`, `client_secret`, etc.

Except for the fix to `openIdConnectUrl` all the tests should pass without modification.

**Comment by samedii** [comment]
I'll make some other PRs then. I think this repo is probably the highest code quality that I've seen on GitHub but I think the design of the library's API feels needlessly complicated. I'll add the PRs you wanted above later today

FYI I am blocked by this and will probably temporarily publish an alternative on pypi with a simpler interface

**Comment by samedii** [comment]
Looking at it again it's not that bad but this stuff 
```
def get_auth(
    *_,
    ...
) -> Callable[[str], IDToken]:
```
in the public interface is so dangerous.

**Pull request closed without merging**